backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt

```diff
@@ -35,4 +44,33 @@ class CardMutationService {
         }
         return true
     }
+
+    @GraphQLDescription("Activate a digital EAK")
+    fun activateCard(
+        project: String,
+        card: CardGenerationModel,
+        overwrite: Boolean
+    ): CardActivationResultModel {
+        if (card.codeType == CodeType.static) {
+            return CardActivationResultModel(ActivationState.failed)
+        }
+
+        val cardHash = Base64.getDecoder().decode(card.cardInfoHashBase64)
+        val activationSecret = Base64.getDecoder().decode(card.activationSecretBase64)
+        val activatedCard = transaction { CardRepository.findByHashModelAndActivationSecret(project, cardHash, activationSecret,) }
+
+        if (activatedCard == null || activatedCard.activationSecret == null) {
+            return CardActivationResultModel(ActivationState.failed)
```

Comment:
I think we can omit the check for `activatedCard.activationSecret == null` here, as we already used the `findByHashAndActivationSecret` function.

We should probably add a log (similar to when logging into administration fails) that the activation failed. (In order to block IPs that try to activate a card by brute forcing it :D )

Reply:
will be done in a separate task.